src/significance.jl

```diff
@@ -0,0 +1,19 @@
+function p_direction(x::Vector{Float64})
+    return maximum([sum(x .> 0) ./ length(x), sum(x .< 0) ./ length(x)])
```

Comment:
Hmm, this creates a few unnecessary allocations (the array inside of `maximum`, `x .> 0`, and `x .< 0`). This could be fixed e.g. by
```suggestion
    ntotal = 0
    npositive = 0
    nnegative = 0
    for xi in x
        if xi > 0
            npositive += 1
        elseif xi < 0
            nnegative += 1
        end
        ntotal += 1
    end
    return max(npositive, nnegative) / ntotal
```

Reply:
I am in disbelief that this code is more efficient than the above one-liner, but I checked with `@btime` and indeed... 🤯

Reply:
The one-liner creates quite a few allocations (zero allocations in the alternative) and it iterates over the array at least twice (one loop in the suggestion).

Reply:
> I am in disbelief that this code is more efficient than the above one-liner, but I checked with `@btime` and indeed... 🤯

If I had a nickle for each time I thought I had written the best code and then David has a better solution -- I'd be so rich